CONTRIBUTING.md

```diff
@@ -1,34 +1,89 @@
 # Contributing
+__We appreciate all kinds of help, so thank you!__
 
-**We appreciate all kinds of help, so thank you!**
 
-## Contributing to Entanglement Forging
+## Code contribution guide
+This guide is for those who want to extend the module or documentation. If you just want to use the package, read [this other guide](./docs/2-reference_guide/reference_guide.md) instead.
 
-Specific details for contributing to this project are outlined below.
+Code in this repository should conform to PEP8 standards. Style/lint checks are run to validate this. Line length must be limited to no more than 100 characters.
 
-### Reporting Bugs and Requesting Features
+### Initial set-up and installing dependencies
+In order to contribute, you will need to [install the module from source](./docs/2-reference_guide/reference_guide.md#installation-from-source). If you do not have write permissions to the original Entanglement Forging repo, you will need to fork it to your personal account first, and submit all Pull Requests (PR) from there. Even if you have write permissions, forking the repo should always work; so this is the recommended approach.
 
-Users are encouraged to use Github Issues for reporting issues are requesting features.
+### Running tests
+If you haven't already, install in editable mode and with developer dependencies:
+```
+pip install -e .[dev]
+```
 
-### Ask/Answer Questions and Discuss Entanglement Forging
+To run tests:
+```
+tox -e{env}
+```
+where you replace `{env}` with `py37`, `py38` or `py39` depending on which version of python you have (to check python version, type `python --version` in the terminal).
 
-Users are encouraged to use Github Discussions for engaging with researchers, developers, and other users regarding this project.
+To run linting tests (checks formatting/syntax):
+```
+tox -elint
+```
 
-### Project Code Style.
+To run all the tests:
+```
+python -m unittest discover -v tests
+```
 
-Code in this repository should conform to PEP8 standards. Style/lint checks are run to validate this. Line length must be limited to no more than 100 characters.
+To run notebook tests (for more info, see [here](https://github.com/ReviewNB/treon)):
+```
+pip install pytest nbmake
```

Comment:
pip install .[dev] instead of explicitly installing things in onesies and twosies. This will help us ensure no dependency conflicts